lib/datadog/core/remote/component.rb

```diff
@@ -20,23 +21,36 @@ def initialize(settings, agent_settings)
           transport_options = {}
           transport_options[:agent_settings] = agent_settings if agent_settings
 
+          negotiation = Negotiation.new(settings, agent_settings)
```

Comment:
I can see the tracing transport needing negotiation to find out what version of the flushing API the agent supports.

Reply:
I kind of agree, although I'm not fond of actual singletons and globals. I was thinking of making it a component, like logging (If you squint you can kind of see it because of `new`'s signature ;)

> I can see the tracing transport needing negotiation to find out what version of the flushing API the agent supports.

Definitely, that is my thought. Actually that was my though even back when I worked on `core/transport` (where you can see I called the transport for `/info` `Negotiation`...)

It's a bigger change, that should be part of the general movement to refactor `ddtrace/transport` and `core/transport` so as a first step I put it isolated in `core/remote` because that was way out of scope for this PR.